biz.aQute.bndlib/src/aQute/bnd/classfile/ClassFile.java

```diff
@@ -153,4 +161,15 @@ static ByteBuffer slice(DataInput in, int length) throws IOException {
 		in.readFully(array, 0, length);
 		return ByteBuffer.wrap(array, 0, length);
 	}
+
+	public byte[] toBytes() throws IOException {
+		ByteBufferDataOutput dout = new ByteBufferDataOutput();
+		write(dout);
+		return dout.toByteArray();
+	}
+
+	public static ClassFile parseFile(File file) throws FileNotFoundException, IOException {
```

Comment:
Don't need the FileNotFoundException since it is a subclass of the already listed IOException.